src/main/groovy/edu/oregonstate/mist/api/Resource.groovy

```diff
@@ -0,0 +1,64 @@
+package edu.oregonstate.mist.api
+
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.Response.ResponseBuilder
+
+/**
+ * Abstract class for reusing common response messages.
+ */
+abstract class Resource {
+    /**
+     * Returns a builder for an HTTP 200 ("ok") response with the argument entity as body.
+     *
+     * @param entity
+     * @return ok response builder
+     */
+    protected static ResponseBuilder ok(Object entity) {
+        ResponseBuilder responseBuilder = Response.ok()
+        responseBuilder.entity(entity)
+    }
+
+    /**
+     * Returns a builder for an HTTP 201 ("created") response with the argument entity as body.
+     *
+     * @param entity
+     * @return created response builder
+     */
+    protected static ResponseBuilder created(Object entity) {
+        ResponseBuilder responseBuilder = Response.status(Response.Status.CREATED)
+        responseBuilder.entity(entity)
+    }
+
+    /**
+     * Returns a builder for an HTTP 400 ("bad request") response with an error message as body.
+     *
+     * @param message
+     * @return bad request response builder
+     */
+    protected static ResponseBuilder badRequest(String message) {
+        ResponseBuilder responseBuilder = Response.status(Response.Status.BAD_REQUEST)
+        responseBuilder.entity(new Error(
+                status: 400,
+                developerMessage: message,
+                userMessage: 'Bad Request',
```

Comment:
Strings such as "Bad Request" and "Not Found" used below should not be defined in the method or even in this class. For strings used in applications or apis, it's best to use a language file. The use of a language file allows us to change strings without making code changes.